File: src/main/java/com/example/chatService/dto/RoomResponse.java
package com.example.chatService.dto;

import com.example.chatService.entity.ChatRoom;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

@Getter
@AllArgsConstructor
@ToString
public class RoomResponse {

    private String roomId;
    private String name;
    private RoomType type;
    private int currentCount;
    private int maxParticipants;

    private boolean accessible;   // 입장 가능 여부
    private String reason;         // PRIVATE_ROOM 등

    private boolean largeRoom;
    private String ownerUserId;
    private String inviteToken; // PRIVATE일 때만

    public static RoomResponse from(ChatRoom r) {
        return new RoomResponse(
                r.getRoomId(),
                r.getName(),
                r.getType(),
                r.getCurrentCount(),
                r.getMaxParticipants(),
                true,
                null,
                r.isLargeRoom(),
                r.getOwnerUserId(),
                null
        );
    }

    public static RoomResponse of(ChatRoom room, String inviteToken) {
        return new RoomResponse(
                room.getRoomId(),
                room.getName(),
                room.getType(),
                room.getCurrentCount(),
                room.getMaxParticipants(),
                true,
                null,
                room.isLargeRoom(),
                room.getOwnerUserId(),
                inviteToken
        );
    }
    public static RoomResponse inaccessible(ChatRoom room, String reason) {
        return new RoomResponse(
                room.getRoomId(),
                room.getName(),
                room.getType(),
                room.getCurrentCount(),
                room.getMaxParticipants(),
                false,
                reason,
                room.isLargeRoom(),
                room.getOwnerUserId(),
                null
        );
    }
}

File: src/main/java/com/example/chatService/dto/RoomResponse.kt
package com.example.chatService.dto

import com.example.chatService.entity.ChatRoom
import kotlin.math.max

data class RoomResponse(
        val roomId: String,
        val name : String,
        val type : RoomType?,
        val currentCount : Int,
        val maxParticipants : Int,

        val accessible : Boolean,
        val reason : String?,
        val largeRoom : Boolean,
        val ownerUserId : String,
        val inviteToken : String?
) {
    companion object {
        fun from(r : ChatRoom) : RoomResponse {
            return RoomResponse(
                    roomId = r.roomId,
                    name = r.name,
                    type = r.type,
                    currentCount = r.currentCount,
                    maxParticipants = r.maxParticipants,
                    accessible = true,
                    reason = null,
                    largeRoom = r.largeRoom,
                    ownerUserId = r.ownerUserId,
                    inviteToken = null
            )
        }
        fun of(r: ChatRoom, inviteToken: String?) : RoomResponse {
            return RoomResponse(
                    roomId = r.roomId,
                    name = r.name,
                    type = r.type,
                    currentCount = r.currentCount,
                    maxParticipants = r.maxParticipants,
                    accessible = true,
                    reason = null,
                    largeRoom = r.largeRoom,
                    ownerUserId = r.ownerUserId,
                    inviteToken = inviteToken
            )
        }
    }
}
